### PR TITLE
Address review comments

### PR DESCRIPTION
**Pull request by sarslanhan**
Based on [review comments](https://gist.github.com/aryszka/441f00024a87fa0da7308f520d83bf2b) from @aryszka

### CODE REVIEW

File: cron.go [hunk 1]
```diff
@@ -21,121 +21,121 @@ var (
...
-	return f.Val == val
+func (f constant) match(val int) bool {
+	return f.val == val
 }
 
-type rangeCronField struct {
-	Start int
-	End   int
+// range is a reserved keyword
+type rangeF struct {
```
Comment (aryszka) [inline cron.go]: like

File: cron.go [hunk 2]
```diff
@@ -21,121 +21,121 @@ var (
...
 
-func (c *cronMask) Matches(t time.Time) bool {
-	return c.Minute.Matches(t.Minute()) &&
-		c.Hour.Matches(t.Hour()) &&
-		c.DayOfMonth.Matches(t.Day()) &&
-		c.Month.Matches(int(t.Month())) &&
-		c.DayOfWeek.Matches(int(t.Weekday()))
+func validateRange(i, val int) error {
+	validRange := validRanges[i]
+	if validRange.start <= val && validRange.end >= val {
```
Comment (aryszka) [inline cron.go]: this is a new question: reasoning about ranges of time can be tricky. Have you considered the lower closed and upper closed approach? As in [0..1) over integers means 0? I don't know what's better, just contemplating which approach can meet the expectation of more users.

Comment (aryszka) [reply cron.go]: mh.. wrong place. The validation is the less relevant place for this pointer, of course.

Comment (sarslanhan) [reply cron.go]: From https://en.wikipedia.org/wiki/Cron#CRON_expression

> Hyphen ( - )
> Hyphens define ranges. For example, 2000–2010 indicates every year between 2000 and 2010, inclusive.

Comment (sarslanhan) [reply cron.go]: I basically tried to drop features that I did not see any use and kept the rest of the spec.

Comment (aryszka) [reply cron.go]: I see. My concern comes from that cron was not so much designed primarily for matching moments in time, but rather for defining moments in time. But our primary use case here is defining time ranges. The problem with the both ends inclusive is that it makes it impossible to define precise time ranges, e.g. two time ranges, night and day, without overlapping. I am still not sure if this is a problem, just asking if it was considered.

File: cron.go [hunk 3]
```diff
@@ -156,29 +156,27 @@ func parseCronField(fieldIdx int, fieldStr string) (cronField, error) {
 // - Command section
 //
 // - Text representation of the fields "month" and "day of week"
-func New(expr string) (CronMask, error) {
+func New(expr string) (*CronMask, error) {
```
Comment (szuecs) [inline cron.go]: I know it's a matter of taste, but I like more the `NewCronMask` style of naming `New`.
Please ignore if you think `New` is better.

Comment (sarslanhan) [reply cron.go]: What I like about having `New` is mostly from user's perspective. To me `cronmask.New` feels better than `cronmask.NewCronMask` as it is already implicit if the library is doing only one thing. What do you think?

Comment (szuecs) [reply cron.go]: makes also sense to me.

File: cron.go (hunk 2 above)
Comment (sarslanhan) [reply cron.go]: We discussed a bit more regarding this. Having the ranges as `[a, b)` has the advantage of being able to represent time ranges without conflicts meaning `0-30 19 * * *` and `30-59 19 * * *` will both match for a minute at `19:30` compared to `[a,b]`. However, `[a,b]` follows the convention from crontab expressions and a more known format. Diverging from the spec has some risk of making the users learn about these exceptional cases.

In either case, I guess we can wait and see how it feels in general and update accordingly in the future.

**Comment by sarslanhan** [comment]
@szuecs @aryszka I will merge this now. If you have anything else in mind, feel free to comment and I will try to update.

File: cron.go (hunk 2 above)
Comment (szuecs) [reply cron.go]: IMO: the risk is to high. The sub part of the name `cron` with the same syntax, but different behavior is not a good idea.

**Pull request merged**